File: WOL_ASPDotNet/Models/ViewModels/PutHostCredentialViewModel.cs
namespace WOL_ASPDotNet.Models.ViewModels
{
    public class PutHostCredentialViewModel
    {
        public string MacAddress { get; set; }

        public int? RDP_Port { get; set; }

        public string RDP_Domain { get; set; }

        public string RDP_UserName { get; set; }

        public string RDP_Password { get; set; }

        public int? SSH_Port { get; set; }

        public string SSH_UserName { get; set; }

        public string SSH_Password { get; set; }

        public int? VNC_Port { get; set; }

        public string VNC_UserName { get; set; }

        public string VNC_Password { get; set; }
    }
}

File: WOL_ASPDotNet/ClientSrc/models/PutHostCredentialViewModel.ts
export type PutHostCredentailViewModel = {
    MacAddress: string,
    RDP_Port: Nullable<number>,
    RDP_Domain: string,
    RDP_UserName: string,
    RDP_Password: string,
    SSH_Port: Nullable<number>,
    SSH_UserName: string,
    SSH_Password: string,
    VNC_Port: Nullable<number>,
    VNC_UserName: string,
    VNC_Password: string
}